Some improvements to makefile

igor725 opened the pull request.

UnknownShadow200 commented:
Tested OpenIndiana (from 2013) does report ```uname -s``` of ```SunOS```

Tested OSX 10.5 does report ```uname -s``` of ```Darwin```

If you're feeling really enthusiastic, I tested some others: NetBSD reports ```NetBSD```, OpenBSD reports ```OpenBSD```,  FreeBSD reports ```FreeBSD```

igor725 commented:
Ok, I will add *BSD a bit later, within the next 4 hours.

igor725 commented:
I think it not works correctly on BSD, tests needed

UnknownShadow200 commented:
make on NetBSD (and similar problem on OpenBSD):
```
make: "/src/Makefile" line 8: Missing dependency operator
make: "/src/Makefile" line 14: Need an operator
make: "/src/Makefile" line 56: Missing dependency operator
make: "/src/Makefile" line 58: Need an operator
```
gmake works fine on both. There's probably no point caring about this.

---
Doing ```tr -s '[:upper:]' '[:lower:]'``` on ```FreeBSD``` produces ```frebsd```, works if ```-s``` is removed. Was there a particular reason for using ```-s```?

FreeBSD also defaults to clang instead of gcc.

Also should be
```
INCDIRS_freebsd=-I /usr/local/include
LDFLAGS_freebsd=-L /usr/local/lib
```
instead of ```INCDIRS_freebsd=-I /usr/local/include -L /usr/Xlocal/lib```

gmake works fine after making these three changes.

igor725 commented:
This Makefile written for GNU Make, BSD Make have slightly different format.
Looks like I misunderstood what the -s option does, I removed it and fixed the variables for freebsd.
